Python/bst.py

```diff
@@ -0,0 +1,222 @@
+            node.right = self._insert(node.right, value)
+
+        node.height = max(self._get_height(node.left), self._get_height(node.right)) + 1
+
+        return node
+
+    def search(self, value):
+        '''Function to search whether the node is present in the BST or not.'''
+        if self.root is None:
+            return False
+        node = self._search(self.root, value)
+        if node:
+            return node
+        else:
+            return False
+
+    def _search(self, node, value):
+        '''Recursive function to search the value in the BST.'''
+
+        # Best case: value is present or if node does not exist
+        if node is None or node.value == value:
+            return node
+
+        # Since AVL Tree follows property of BST, if value is lesser than current node's value,
+        # it will be present in left sub tree.
+        elif value < node.value:
+            return self._search(node.left, value)
+
+        # If value is greater than current node's value then it will be present in right sub tree.
+        else:
+            return self._search(node.right, value)
+
+    def delete(self, value):
+        '''Function to delete a value from the BST.'''
+        self.root = self._delete(self.root, value)
+
+    def _delete(self, node, value):
+        '''Recursive function to delete a node from the BST.'''
+        if node is None:
+            return node
+        elif value < node.value:
+            node.left = self._delete(node.left, value)
+        elif value > node.value:
+            node.right = self._delete(node.right, value)
+        else:
+            # Node has right child present or it does not have both childs
+            if node.left is None:
+                temp = node.right
+                node = None
+                return temp
+
+            # Node has left child present.
+            elif node.right is None:
+                temp = node.left
+                node = None
+                return temp
+
+            # Node has both child present
+            # Get the in-order successor element in the right subtree, and replace
+            # it by the node to be deleted and delete the in-order successor.
+            temp = self._get_inorder_successor(node.right)
+            node.value = temp.value
+            node.right = self._delete(node.right, temp.value)
+
+        # Update the heights of the node after deletion
+        node.height = 1 + max(self._get_height(node.left), self._get_height(node.right))
+
+        # Rebalance the node
+        return node
+
+    def _get_inorder_successor(self, node):
+        '''Returns the minimum node in the tree from a given node.'''
+        if node is None or node.left is None:
+            return node
+        return self._get_inorder_successor(node.left)
+
+    def in_order(self):
+        in_order_list = []
+        self._in_order(self.root, in_order_list)
+        return in_order_list
+
+    def _in_order(self, node, in_order_list):
```

Comment:
```suggestion
    def in_order(self, node, in_order_list=[]):
```

Do this for all operations.

Reply:
How do I return the list, after making these changes ?